iaso/api/profiles/profile_logs.py

```diff
@@ -137,10 +139,53 @@ def get_fields_modified(self, modification):
 
 
 class ProfileLogRetrieveSerializer(serializers.ModelSerializer):
+    past_value = serializers.SerializerMethodField(read_only=True)
+    new_value = serializers.SerializerMethodField(read_only=True)
+
     class Meta:
         model = Modification
         fields = ["id", "created_at", "user", "source", "new_value", "past_value", "object_id", "content_type"]
 
+    def get_past_value(self, modification):
+        logged_org_units = modification.past_value[0]["fields"]["org_units"]
+        logged_projects = modification.past_value[0]["fields"]["projects"]
+        logged_user_roles = modification.past_value[0]["fields"]["user_roles"]
+
+        org_units_names_and_ids = OrgUnit.objects.filter(id__in=logged_org_units).values("id", "name")
+        projects_names_and_ids = Project.objects.filter(id__in=logged_projects).values("id", "name")
+        user_roles_group_names_and_ids = UserRole.objects.filter(id__in=logged_user_roles).values("id", "group__name")
+
+        user_roles_names_and_ids = [
+            {"id": entry["id"], "name": UserRole.remove_user_role_name_prefix(entry["group__name"])}
+            for entry in user_roles_group_names_and_ids
+        ]
+
+        past_value_copy = copy.deepcopy(modification.past_value)
+        past_value_copy[0]["fields"]["org_units"] = org_units_names_and_ids
+        past_value_copy[0]["fields"]["projects"] = projects_names_and_ids
+        past_value_copy[0]["fields"]["user_roles"] = user_roles_names_and_ids
+
+        return past_value_copy
+
+    def get_new_value(self, modification):
+        logged_org_units = modification.new_value[0]["fields"]["org_units"]
+        logged_projects = modification.new_value[0]["fields"]["projects"]
+        logged_user_roles = modification.new_value[0]["fields"]["user_roles"]
+
+        org_units_names_and_ids = OrgUnit.objects.filter(id__in=logged_org_units).values("id", "name")
```

Comment:
Since fetching IDs and names is the same for both past and new value, I would put the fetching and transformation in a utility function and call it in both `get_xxx_value()`